packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -629,6 +1071,329 @@ class _NoDefaultValue {
+///
+/// See also:
+///
+/// * [DiagnosticsNode.toStringDeep], which uses a [TextRender] to return a
+///    string representation of this node and its descendants.
+class TextTreeRenderer {
+  /// Creates a [TextTreeRenderer] object with the given arguments specifying
+  /// how the tree is rendered.
+  ///
+  /// Lines are wrapped to at the maximum of [wrapWidth] and the current indent
+  /// plus [wrapWidthProperties] characters. This ensures that wrapping does not
+  /// become too excessive when displaying very deep trees and that wrapping
+  /// only occurs at the overall [wrapWidth] when the tree is not very indented.
+  /// If [maxDescendentsTruncatableNode] is specified, [DiagnosticsNode] objects
+  /// with `allowTruncate` set to `true` are truncated after including
+  /// [maxDescendentsTruncatableNode] descendents of the node to be truncated.
+  TextTreeRenderer({
+    DiagnosticLevel minLevel = DiagnosticLevel.debug,
+    int wrapWidth = 100,
+    int wrapWidthProperties = 65,
+    int maxDescendentsTruncatableNode = -1,
+  }) : assert(minLevel != null),
+       _minLevel = minLevel,
+       _wrapWidth = wrapWidth,
+       _wrapWidthProperties = wrapWidthProperties,
+       _maxDescendentsTruncatableNode = maxDescendentsTruncatableNode;
+
+  final int _wrapWidth;
+  final int _wrapWidthProperties;
+  final DiagnosticLevel _minLevel;
+  final int _maxDescendentsTruncatableNode;
+
+  /// Text configuration to use to connect this node to a `child`.
+  ///
+  /// The singleLine styles are special cased because the connection from the
+  /// parent to the child should be consistent with the parent's style as the
+  /// single line style does not provide any meaningful style for how children
+  /// should be connected to their parents.
+  TextTreeConfiguration _childTextConfiguration(
+      DiagnosticsNode child,
+      TextTreeConfiguration textStyle,
+  ) {
+    final DiagnosticsTreeStyle childStyle = child?.style;
+    return (_isSingleLine(childStyle) || childStyle == DiagnosticsTreeStyle.indentedSingleLine) ? textStyle : child.textTreeConfiguration;
+  }
+
+  /// Renders a [node] to a String.
+  String render(
+    DiagnosticsNode node, {
+    String prefixLineOne = '',
+    String prefixOtherLines,
+    TextTreeConfiguration parentConfiguration,
+  }) {
+    final bool isSingleLine = _isSingleLine(node.style) && parentConfiguration?.lineBreakProperties != true;
+    prefixOtherLines ??= prefixLineOne;
+    if (node.linePrefix != null) {
+      prefixLineOne += node.linePrefix;
+      prefixOtherLines += node.linePrefix;
+    }
+
+    final TextTreeConfiguration config = node.textTreeConfiguration;
+    if (prefixOtherLines.isEmpty)
+      prefixOtherLines += config.prefixOtherLinesRootNode;
+
+    if (node.style == DiagnosticsTreeStyle.truncateChildren) {
+      // This style is different enough that it isn't worthwhile to reuse the
+      // existing logic.
+      final List<String> descendants = <String>[];
```

Comment:
Maybe prefer "descendents" to be consistent w/ elsewhere?

Reply:
Actually across all of flutter descendants is the standard and I was the outlier. Switched to descendants everywhere in this file for consistency.

Reply:
"Descendant is both an adjective (meaning either moving downward or descending from an ancestor) and a noun (for someone who descends from someone else). Descendent is a less common variant used indiscriminately in place of descendant in all its uses."